main/src/ca/uwaterloo/flix/language/phase/Verifier.scala

```diff
@@ -426,23 +402,45 @@ object Verifier {
   private def checkEq(tpe1: MonoType, tpe2: MonoType, loc: SourceLocation): MonoType = {
     if (tpe1 == tpe2)
       tpe1
-    else
-      throw MismatchedTypes(tpe1, tpe2, loc)
+    else failMismatchedTypes(tpe1, tpe2, loc)
   }
 
   /**
-    * An exception raised because `tpe` did not match an expected shape
+    * Throw `InternalCompilerException` because the `found` type does not match the shape specified by `expected`
     */
-  private case class MismatchedShape(tpe: MonoType, expected: String, loc: SourceLocation) extends RuntimeException
+  private def failMismatchedShape(found: MonoType, expected: String, loc: SourceLocation): Nothing =
+    throw InternalCompilerException(
+      s"""Mismatched shape near ${loc.format}
```

Comment:
Unfortunately such nice multiline error messages are probably not printed correctly.

Can you refactor and have a single line error message? Thanks!